### ISSUE TITLE
Should there be an extra context cache clearing in `multistatesampler.__del__`?

### ISSUE THREAD

**Issue by IAlibay** [choderalab/openmmtools#689]
Context: From today's OpenFE call, @mikemhenry recommended opening this just to review later.

Quite simply, should there be explit calls to clearing the context cache in `__del__` for multistatesampler?

**Comment by jchodera** [comment]
Yes! We just hadn't run into this before because we don't typically run multiple `MultiStateSampler` calls in a row, but this may have also been the cause of other issues we had seen earlier.

cc https://github.com/choderalab/openmmtools/issues/688

**Comment by ijpulidos** [comment]
Trying to revisit this issue. I think we definitely want to do this, but we should come up with a test case for this. Is there a script somewhere that we can use a starting point for the test? Thanks!
